lms/services/file.py

```diff
@@ -13,32 +13,39 @@ def __init__(self, application_instance, db):
 
     def get(self, lms_id, type_, course_id=None) -> Optional[File]:
         """Return the file with the given parameters or None."""
-        return self._file_search_query(
-            application_instance=self._application_instance,
-            lms_id=lms_id,
-            type_=type_,
-            course_id=course_id,
-        ).one_or_none()
+        return (
+            self._file_search_query(
+                guid=self._application_instance.tool_consumer_instance_guid,
+                lms_id=lms_id,
+                type_=type_,
+                course_id=course_id,
+            )
+            # We might have recorded the same file twice in different ApplicationInstances
+            # as we query by GUID we might get more than once row, in that case we prefer the newest
+            .order_by(File.id.desc()).first()
```

Comment:
Yeah, pretty much, but to be pedantically correct:

>  Also it looks like the query returned by _file_search_query() can now return multiple results

`_file_search_query` returns the raw SQLA query so it could return none, one or multiple results depending of the parameters. The main change is that for the minimum set of query params, the change to query by guid instead of AIs. means that in those case it can return multiple when before it could only return one. 

The callers need to change the SQLA method they use (basically stop using any that has `one`)

Reply:
Yeah, pretty much, but to be pedantically correct:

>  Also it looks like the query returned by _file_search_query() can now return multiple results

`_file_search_query` returns the raw SQLA query so it could return none, one or multiple results depending of the parameters. The main change is that for the minimum set of query params, the change to query by guid instead of AIs. means that in those case it can return multiple when before it could only return one. 

The callers need to change the SQLA method they use (basically stop using any that has `one`)